### DISCUSSION TITLE
Add a dropout layer

### DISCUSSION THREAD

**Discussion by sallem** [opendilab/DI-engine#705] [Q&A] [answered]
Hey, I would like to add dropout layers into mu DQN model.

A dropout is a layer useful for training a more robust model by randomly removing some values during the training in order to generalize it.

Is it possible to add it with the config file ? Did not find this information into the documentation.

**Comment by karroyan** [comment, marked as answer]
Hi, thank you for raising this issue. 

We plan to implement the ability to add layers via the config file within this week. We will open a pull request once this feature is ready. Please keep an eye on the repository for updates.

**Comment by karroyan** [comment]
@sallem, hi, in which environments you plan to use DQN?

**Reply by sallem** [reply]
Hey

I am using it with stocks environment for trading.

**Comment by karroyan** [comment]
Hi, the dropout control has been implemented in pull request #712 . You can now add a ``dropout=0.3`` parameter under the ``config.policy.model`` configuration to enable dropout regularization with dropout probability of 0.3 after each linear layer followed by activation in DQN with FCEncoder. The configuration example can be found in [cartpole_dqn_config](https://github.com/opendilab/DI-engine/blob/main/dizoo/classic_control/cartpole/config/cartpole_dqn_config.py). Let me know if you have any other questions!
